testsuite/smoke-test/src/jwks/dummy_provider/request_handler.rs

```diff
@@ -0,0 +1,114 @@
+// Copyright © Aptos Foundation
+
+use aptos_infallible::Mutex;
+use hyper::{Body, Request};
+use move_core_types::account_address::AccountAddress;
+use std::{collections::HashSet, str::FromStr};
+
+/// A handler that handles JWK requests from a validator,
+/// assuming the validator account address is written as the COOKIE.
+pub trait RequestHandler: Send + Sync {
+    fn handle(&self, request: Request<Body>) -> Vec<u8>;
+}
+
+pub struct StaticContentServer {
+    content: Vec<u8>,
+}
+
+impl StaticContentServer {
+    pub fn new(content: Vec<u8>) -> Self {
+        Self { content }
+    }
+}
+
+impl RequestHandler for StaticContentServer {
+    fn handle(&self, _origin: Request<Body>) -> Vec<u8> {
+        self.content.clone()
+    }
+}
+
+fn origin_from_cookie(request: &Request<Body>) -> AccountAddress {
+    let cookie = request
+        .headers()
+        .get(hyper::header::COOKIE)
+        .unwrap()
+        .to_str()
+        .unwrap();
+    AccountAddress::from_str(cookie).unwrap()
+}
+
+/// The first `k` requesters will get content A forever, the rest will get content B forever.
+pub struct EquivocatingServer {
+    content_a: Vec<u8>,
+    content_b: Vec<u8>,
+    k: usize,
+    requesters_observed: Mutex<HashSet<AccountAddress>>,
+}
+
+impl EquivocatingServer {
+    pub fn new(content_a: Vec<u8>, content_b: Vec<u8>, k: usize) -> Self {
+        Self {
+            content_a,
+            content_b,
+            k,
+            requesters_observed: Mutex::new(HashSet::new()),
+        }
+    }
+}
+
+impl RequestHandler for EquivocatingServer {
+    fn handle(&self, request: Request<Body>) -> Vec<u8> {
+        let mut requesters_observed = self.requesters_observed.lock();
+        let origin = origin_from_cookie(&request);
```

Comment:
cc @grao1991 that's why the `cfg` in `jwk_observer`. I couldn't figure out a better way to do implement equivocation.